gr-digital/python/digital/qa_diff_stbc_decoder_cc.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# 
+# Copyright 2018 Free Software Foundation, Inc.
+# 
+# This file is part of GNU Radio
+# 
+# GNU Radio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+# 
+# GNU Radio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with GNU Radio; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+# 
+
+from gnuradio import gr, gr_unittest
+from gnuradio import blocks
+import digital_swig as digital
+import pmt
+import numpy as np
+
+M_SQRT_2 = 1.0/np.sqrt(2)
+
+class qa_diff_stbc_decoder_cc (gr_unittest.TestCase):
+
+    def setUp (self):
+        self.tb = gr.top_block ()
+
+    def tearDown (self):
+        self.tb = None
+
+    # Produce stream tags and calculate expected result.
+    def decode(self, basis, input, tag_pos):
+        # Calculate the expected behaviour without the presence of tags.
+        output = np.empty(shape=[len(input)], dtype=complex)
+        data = np.append(basis, input)
+        output[::2] = (data[2::2]*np.conj(data[0:len(input):2]) + np.conj(data[3::2])*data[1:len(input):2])*basis[0] - \
+                      (data[2::2]*np.conj(data[1:len(input):2]) - np.conj(data[3::2])*data[0:len(input):2])*np.conj(basis[1])
+        output[1::2] = (data[2::2]*np.conj(data[0:len(input):2]) + np.conj(data[3::2])*data[1:len(input):2])*basis[1] + \
+                       (data[2::2]*np.conj(data[1:len(input):2]) - np.conj(data[3::2])*data[0:len(input):2])*np.conj(basis[0])
+
+        # Iterate over tags and update the calculated output according to the diced CSI.
```

Comment:
Just because you write this at multiple locations: "to dice" sounds a little odd to me :). Why not just "random CSI"?